web/cypress/integration/route/create-route-with-upstream.spec.js

```diff
@@ -0,0 +1,142 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable no-undef */
+
+context('Create Route with Upstream', () => {
+  const domSelector = {
+    input: ':input',
+    notification: '.ant-notification-notice-message',
+    node_0_host: '#nodes_0_host',
+    name: '#name',
+    description: '#desc',
+    upstreamSelector: '#upstream_id',
+    searchName: '[title=Name]',
+  };
+
+  const data = {
+    upstreamName: 'test_upstream',
+    routeName: 'test_route',
+    description: 'desc_by_autotes',
+    host: '10.89.90.237',
+    ip1: '127.0.0.1',
+    ip2: '127.0.0.2',
+    deleteRouteSuccess: 'Delete Route Successfully',
+    deleteUpstreamSuccess: 'Delete successfully',
+  };
+
+  beforeEach(() => {
+    cy.login();
+  });
+
+  it('should create an upstream', () => {
+    cy.visit('/');
+    cy.contains('Upstream').click();
+    cy.contains('Create').click();
+
+    cy.get(domSelector.name).type(data.upstreamName);
+    cy.get(domSelector.description).type(data.description);
+    cy.get(domSelector.node_0_host).type(data.host);
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+  });
+
+  it('should enter the Route creator', () => {
+    cy.visit('/');
+    cy.contains('Route').click();
+    cy.contains('Create').click();
+
+    cy.get(domSelector.name).type(data.routeName);
+    cy.contains('Next').click();
+  });
+
+  it('should disable Upstream input boxes after selecting an existing upstream', () => {
+    cy.contains('Custom').click();
+    cy.contains(data.upstreamName).click();
+    cy.get(domSelector.input).should('be.disabled');
+  });
+
+  it('should enable Upstream input boxes after selecting Custom mode', () => {
+    cy.contains(data.upstreamName).click();
+    cy.contains('Custom').click();
+    cy.get(domSelector.input).should('not.be.disabled');
+  });
+
+  it('should submit custom Upstream properties successfully', () => {
+    cy.get(domSelector.node_0_host).clear().type(data.ip1);
+    cy.contains('Next').click();
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+    cy.contains('Goto List').click();
+    cy.url().should('contains', 'routes/list');
+  });
+
+  it('should edit this route with upstream', () => {
+    cy.visit('/');
+    cy.contains('Route').click();
+
+    cy.get(domSelector.searchName).type(data.routeName);
+    cy.contains('Search').click();
+    cy.contains(data.routeName).siblings().contains('Edit').click();
+
+    // fetching Resource
+    cy.wait(500);
```

Comment:
https://github.com/apache/apisix-dashboard/pull/1304/files#diff-581f75d9ca2eb23441d82c276047d55b481a8399ea296834decad24b508c69eaR23

Use this to check if that request finished.